traitsui/tests/editors/test_tabular_editor.py

```diff
@@ -39,30 +45,326 @@ class Report(HasTraits):
+            self.assertEqual(report.multi_selected, [])
+
+            set_selected_multiple(editor, [0, 1])
+            gui.process_events()
+
+            self.assertEqual(report.selected_rows, [0, 1])
+            self.assertEqual(report.multi_selected, people[:2])
+
+            set_selected_multiple(editor, [2])
+            gui.process_events()
+
+            self.assertEqual(report.selected_rows, [2])
+            self.assertEqual(report.multi_selected, [people[2]])
+
+            clear_selection(editor)
+            gui.process_events()
+
+            self.assertEqual(report.selected_rows, [])
+            self.assertEqual(report.multi_selected, [])
+
+    @unittest.skipIf(is_current_backend_wx(), "Issue enthought/traitsui#752")
+    def test_tabular_editor_single_selection_changed(self):
+        gui = GUI()
+
+        with store_exceptions_on_all_threads(), \
+                self.report_and_editor(get_view()) as (report, editor):
+            gui.process_events()
+            people = report.people
+
+            self.assertEqual(get_selected_rows(editor), [])
+
+            report.selected_row = 1
+            gui.process_events()
+
+            self.assertEqual(get_selected_rows(editor), [1])
+            self.assertEqual(report.selected, people[1])
+
+            report.selected = people[2]
+            gui.process_events()
+
+            self.assertEqual(get_selected_rows(editor), [2])
+            self.assertEqual(report.selected_row, 2)
+
+            # Selected set to invalid value doesn't change anything
+            report.selected = Person(name="invalid", age=-1)
+            gui.process_events()
+
+            self.assertEqual(get_selected_rows(editor), [2])
+            self.assertEqual(report.selected_row, 2)
+
+            # -1 clears selection
+            report.selected_row = -1
+            gui.process_events()
+
+            self.assertEqual(get_selected_rows(editor), [])
+            self.assertEqual(report.selected, None)
+
+    @unittest.skipIf(is_current_backend_wx(), "Issue enthought/traitsui#752")
+    def test_tabular_editor_multi_selection_changed(self):
+        gui = GUI()
+        view = get_view(multi_select=True)
+
+        with store_exceptions_on_all_threads(), \
+                self.report_and_editor(view) as (report, editor):
+            gui.process_events()
+            people = report.people
+
+            self.assertEqual(get_selected_rows(editor), [])
+
+            report.selected_rows = [0, 1]
+            gui.process_events()
+
+            self.assertEqual(get_selected_rows(editor), [0, 1])
+            self.assertEqual(report.multi_selected, people[:2])
+
+            report.multi_selected = [people[2], people[0]]
+            gui.process_events()
+
+            self.assertEqual(sorted(get_selected_rows(editor)), [0, 2])
+            self.assertEqual(sorted(report.selected_rows), [0, 2])
+
+            # If there's a single invalid value, nothing is updated
```

Comment:
Nit: This could be a separate unit test such that assertion failures above this line won't prevent this test from being run, which is probably more helpful for developers when making changes to the code being tested.

Reply:
The argument could apply to every single assertion group in this and other tests. I don't see anything special about this group so I would like to keep the test as is -- it's a reasonable size test that checks the behaviour of one specific function. And if somebody is working on that part of the code, they should make sure the whole test passes anyways.

Reply:
The reason why I saw this block being separate is because it tests a separate code branch to the ones tested above, i.e. the `except` branch, verse the `else` branch in `_multi_selected_changed`.

Reply:
It is okay keeping this test as is, which is why I mark this as Nit.

Reply:
The same argument would still apply to more than these couple locations, so I'll keep them the same.